core/src/main/java/io/questdb/cutlass/text/CairoTextWriter.java

```diff
@@ -175,11 +193,24 @@ public void onFieldsPartitioned(long line, ObjList<DirectByteCharSequence> value
                 if (onField(line, dbcs, w, i)) return;
             }
             w.append();
+            checkMaxAndCommitLag();
         } catch (Exception e) {
             logError(line, timestampIndex, dbcs);
         }
     }
 
+    private void checkMaxAndCommitLag() {
+        if (writer != null) {
+            if (durable) {
```

Comment:
I'd suggest to invert condition for better optimisation so that more often hit branch is first